doc/kernel/other/memory_domain.rst

```diff
@@ -0,0 +1,192 @@
+.. _memory_domain:
+
+Memory Domain
+#############
+
+The memory domain APIs are used by unprivileged threads to share data to
+the threads in the same memory and protect sensitive data from threads
+outside their domain. It is not only for improving the security but also
```

Comment:
change "It is..." to "Memory domains are not only used for improving security, but are also"